src/indexer.rs

```diff
@@ -611,3 +611,63 @@ impl Indexer for AffectedContractIndexer {
         false
     }
 }
+
+/// An indexer that retrieves all events in a block, transaction outcomes
+/// and special transaction outcomes.
+///
+/// The [`on_connect`](Indexer::on_connect) and
+/// [`on_failure`](Indexer::on_failure) methods of the [`Indexer`] trait only
+/// log the events on `info` and `warn` levels, respectively, using the
+/// [`tracing`](https://docs.rs/tracing/latest/tracing/) crate. The [target](https://docs.rs/tracing/latest/tracing/struct.Metadata.html#method.target)
+/// of the log is `ccd_indexer` which may be used to filter the logs.
+pub struct BlockEventsIndexer;
+
+#[async_trait]
+impl Indexer for BlockEventsIndexer {
+    type Context = ();
+    type Data = (
```

Comment:
**Suggestion (non-blocking):** Consider introducing a struct for this.

Reply:
I considered it, but since the types are all different this seems more convenient also for using. So won't do this.